client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -165,9 +165,9 @@ public Response performRequest(String method, String endpoint, Map<String, Strin
 
     /**
      * Sends a request to the Elasticsearch cluster that the client points to and waits for the corresponding response
-     * to be returned. Shortcut to {@link #performRequest(String, String, Map, HttpEntity, HttpAsyncResponseConsumer, Header...)}
-     * which doesn't require specifying an {@link HttpAsyncResponseConsumer} instance, {@link HeapBufferedAsyncResponseConsumer}
-     * will be used to consume the response body.
+     * to be returned. Shortcut to {@link #performRequest(String, String, Map, HttpEntity, HttpAsyncResponseConsumerFactory, Header...)}
+     * which doesn't require specifying an {@link HttpAsyncResponseConsumerFactory} instance,
+     * {@link HttpAsyncResponseConsumerFactory} will be used to create the needed instances of {@link HttpAsyncResponseConsumer}.
```

Comment:
Maybe "Shortcut to {@link #performRequest(String, String, Map, HttpEntity, HttpAsyncResponseConsumerFactory, Header...)} which defaults the {@link HttpAsyncResponseConsumerFactory} to {@link HttpAsyncResponseConsumerFactory#HEAP_BUFFER_MAX_20MB} which buffers responses from Elasticsearch on a heap, failing the request if the response is larger than that. Consider providing a {@link HttpAsyncResponseConsumerFactory} with a higher limit if scrolling." ?